common/test/components/TextInput.test.js

```diff
@@ -0,0 +1,78 @@
+import { useState } from 'react';
+import TextInput from '../../views/components/TextInput/TextInput';
+import { mountWithTheme } from '../fixtures/enzyme-helpers';
+import useValidation from '../../hooks/useValidation';
+
+const ExampleTextInput = () => {
+  const [value, setValue] = useState('');
+
+  return (
+    <TextInput
+      type="email"
+      value={value}
+      handleChange={event => setValue(event.currentTarget.value)}
+      required={true}
+      errorMessage={`test error message`}
+      handle
+      {...useValidation()}
+    />
+  );
+};
+
+describe('TextInput', () => {
+  test(`The input will validate when blurred`, () => {
+    const wrapper = mountWithTheme(<ExampleTextInput />);
+    const input = wrapper.find('input');
+
+    input.simulate('focus');
+    input.instance().value = 'invalidemail@';
+    input.simulate('change');
+    input.simulate('blur');
+
+    expect(
+      wrapper.find("[data-test-id='TextInputErrorMessage']")
+    ).toMatchSnapshot();
+  });
+  test(`The input will hide errors at the point it becomes valid`, () => {
+    const wrapper = mountWithTheme(<ExampleTextInput />);
+    const input = wrapper.find('input');
+
+    input.simulate('focus');
+    input.instance().value = 'invalidemail@';
+    input.simulate('change');
+    input.simulate('blur');
+
+    expect(
+      wrapper.find("[data-test-id='TextInputErrorMessage']")
+    ).toMatchSnapshot();
+
+    input.simulate('focus');
+    input.instance().value = 'valid@email.com';
+    input.simulate('change');
+
+    expect(
+      wrapper.find("[data-test-id='TextInputErrorMessage']")
```

Comment:
Isn't there a better way to do this? Could we not search for a component or text?

Reply:
I don't _think_ we can search for StyledComponents within a component (will check). We settled on `data-test-id` attributes in previous tests to accomplish this. The trade-off is that we have an extra attribute on the element in production, but hard to see any real harm in this?